pysat/_meta.py

```diff
@@ -1465,18 +1469,48 @@ def default_values_from_type(self, val_type):
         default_val : str, float, int, NoneType
             Sets NaN for all float values, -1 for all int values, and '' for
             all str values except for 'scale', which defaults to 'linear', and
-            None for any othere data type
+            None for any other data type
 
         """
 
+        # Perform some pre-checks on type, checks that could error with
+        # unexepected input.
+        try:
+            floating_check = isinstance(val_type(), np.floating)
+        except TypeError as err:
+            if str(err).find('not a callable function') > 0:
+                floating_check = False
+            else:
+                # Unexpected input
+                floating_check = None
```

Comment:
I don't think this works.  Example:
```
import numpy as np
val_type = np.asarray([15]).astype(np.float64)[0]
type(val_type)
numpy.float64

floating_check = isinstance(val_type(), np.floating)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-23-4e16dfc35319> in <module>
----> 1 floating_check = isinstance(val_type(), np.floating)

TypeError: 'numpy.float64' object is not callable
```

This yields `floating_check = False`.  Following:
```
val_type is float or floating_check
False
```
Which is not really what you were going for, I believe.

Reply:
`val_type` in your example is a float number. The `default_vals_from_type` function is used when a meta variable is created and looks to the MetaLabels class. It is expecting an input like `float` or `np.float64`.

`labels = pysat._meta.MetaLabels(max_val=('max_val', float), min_val=('min_val', np.float64))`

If a user supplies a number instead then:
```
In [11]: meta = pysat.Meta(labels={'min_val':('min_val', 17)})                                                                                                                                                              

In [12]: meta                                                                                                                                                                                                               
Out[12]: 
pysat.Meta(metadata=Empty DataFrame
Columns: [units, long_name, notes, desc, min_val, value_max, fill]
Index: [], labels=pysat.MetaLabels(units=units <class 'str'>, name=long_name <class 'str'>, notes=notes <class 'str'>, desc=desc <class 'str'>, min_val=min_val 17, max_val=value_max <class 'float'>, fill_val=fill <class 'float'>)export_nan=[]) -> 0 Variables

In [13]: meta['hi'] = {'units': 'time'}                                                                                                                                                                                     
pysat INFO: No type match found for 17

```

Reply:
Latest commits updates this to:
```
In [1]: import pysat                                                                                                                                                                                                        

In [2]: pysat.logger.setLevel(1)                                                                                                                                                                                            

In [3]: meta = pysat.Meta(labels={'min_val':('min_val', 17)})                                                                                                                                                               

In [4]: meta['hi'] = {'units': 'time'}                                                                                                                                                                                      
pysat INFO: No type match found for 17
pysat INFO: A problem may have been encountered with the user supplied type for Meta attribute:  min_val Please check the settings provided to `labels` at Meta instantiation.
```

Reply:
🐶  Well that wasn't the smartest by me.  Checked again with the right input type and it works.

Reply:
no worries 😄